modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -42,9 +45,22 @@ sealed class IO<out A> : IOOf<A> {
 
     fun <A> raiseError(e: Throwable): IO<A> = RaiseError(e)
 
-    operator fun <A> invoke(f: () -> A): IO<A> = defer { Pure(f()) }
+    operator fun <A> invoke(ctx: CoroutineContext, f: suspend () -> A): IO<A> =
+      effect(ctx, f)
 
-    fun <A> defer(f: () -> IOOf<A>): IO<A> = Suspend(f)
+    operator fun <A> invoke(f: suspend () -> A): IO<A> =
```

Comment:
How about something like:
```suggestion
    operator fun <A> invoke(f: suspend IO.Companion.() -> A): IO<A> =
```